### PR TITLE
Fixed gettext rake task for plugins

### PR DESCRIPTION
**Pull request by lzap**
When extracting strings for plugins via

    be rake plugin:gettext[foreman_discovery]

the extension in tasks creates irrelevant file in

    foreman_discovery/locale/action_names.rb

which is then extracted. This avoids that.

### CODE REVIEW

**Comment by adamruzicka** [comment]
@lzap Needs a rebase

**Comment by lzap** [comment]
Rebased.

File: lib/tasks/gettext.rake [hunk 1]
```diff
@@ -7,16 +7,13 @@ gettext_find_task = begin
...
-                 .all_action_names
-                 .uniq
-                 .map { |n| %[_("#{n}")] }
-                 .join("\n")
+      # not relevant when extracting plugins
+      if text_domain == "foreman"
+        storage_file = "#{locale_path}/action_names.rb"
+        puts "writing action translations to: #{storage_file}"
+        File.write storage_file, "# Autogenerated!\n" +
+          Actions::EntryAction. all_action_names. uniq. map { |n| %[_("#{n}")] }. join("\n")
```
Comment (adamruzicka) [inline lib/tasks/gettext.rake]:
```
Offenses:

lib/tasks/gettext.rake:15:11: C: Align the operands of an expression spanning multiple lines.
          Actions::EntryAction. all_action_names. uniq. map { |n| %[_("#{n}")] }. join("\n")
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by lzap** [comment]
Ok bot try now.

**Comment by adamruzicka** [comment]
[test]

File: lib/tasks/gettext.rake [hunk 2]
```diff
@@ -7,16 +7,13 @@ gettext_find_task = begin
...
-      puts "writing action translations to: #{storage_file}"
-
-      File.write storage_file,
-                 "# Autogenerated!\n" +
-                 Actions::EntryAction
-                 .all_action_names
-                 .uniq
-                 .map { |n| %[_("#{n}")] }
-                 .join("\n")
+      # not relevant when extracting plugins
```
Comment (iNecas) [inline lib/tasks/gettext.rake]: This is not valid: it's relevant for plugins, that define actions for foreman-tasks (see https://github.com/Katello/katello/blob/master/locale/action_names.rb),

Better approach would be first introduce a mechanism that plugins that use the actions would use for extracting just names of their actions, and then removing this code completely. Without doing this, this PR just introduces regression IMO

**Pull request closed without merging**